client/my-sites/plans-grid/components/header-price.tsx

```diff
@@ -128,7 +128,7 @@ const HeaderPriceContainer = styled.div`
 const PlanFeatures2023GridHeaderPrice = ( {
 	planSlug,
 	isLargeCurrency,
-	isPlanUpgradeCreditEligible,
+	proRatedCreditsApplicable,
```

Comment:
> Any reason we're switching

p.s. you obviously knew or could guess the reason, right? 🤔

Reply:
> Any reason we're switching

Not actually, otherwise I wouldn't ask the question? 🤔  

>I like it as the internal nomenclature

>I don't find that any better,

Having a different nomenclature internally/externally can be confusing specially when considering long term maintainability of code. Since our code and the documented understanding that our users see our inconsistent. This is specially important when considering a new developer who has to absorb this context. There is a gap in the `ubiquitous language`. Do you not think that closing this gap as much as possible is [important](https://tanzu.vmware.com/developer/blog/ubiquitous-language/)?

Reply:
> Not actually, otherwise I wouldn't ask the question? 🤔

That's interesting - I had the impression this is a known and common term used to describe the scenario with upgrade credits. I have seen numerous mentions, it made it into recent docs for plans, discussions, public mentions, it's been quite central for the plans work (pricing and packaging project work in particular had a few iterations on clarifying its meaning/use e.g. pdgrnI-24d-p2), etc. It's a term that dates back quite a while - but if you've never heard of it, then I can't quite argue with that. I never really paid attention either, but accepted it as such.

I don't object be happy to refactor "proRatedCredit" -> "planUpgradeCredit". Just not super convinced tbh.

Reply:
So the more I dig, the more mention I find of this term both internally and externally e.g. these are from happy chat bot logs I spot now in Slack:

> When you initially sign up for a plan and then decide to upgrade to a higher tier within the same billing cycle, WordPress applies what's known as a "**pro-rated credit**" for the time unused on your current plan. This works by calculating the unused value of your current plan and then deducting that amount from the cost of the plan you're upgrading to. This is likely why you originally saw a lower price of C$276 for the Business plan.

> After you complete the payment, your plan will be downgraded to the Personal plan instantly and the **pro-rated** amount of the Business plan will be refunded.

Given this is such a peculiar specification of "upgrade credits" (which by definition is more general) - recall a recent discussion in Slack now p1694467856357549/1694142911.437209-slack-CFFF01Q4V - I'm thinking it might actually be more beneficial now to keep the term in the code, as I have done in this PR. If it helps to highlight or spark curiosity about these conditions related to the credits, then that's probably worth it for anyone coding around these parts.

I would even argue that "upgrade credits" shown in the notice/banner might benefit equally from being renamed to "pro-rated credits".

Reply:
> Given this is such a peculiar specification of "upgrade credits" (which by definition is more general) - recall a recent discussion in Slack now p1694467856357549/1694142911.437209-slack-CFFF01Q4V - I'm thinking it might actually be more beneficial now to keep the term in the code, as I have done in this PR. If it helps to highlight or spark curiosity about these conditions related to the credits, through a common/canonical and apparently adopted term, then that's probably worth it for anyone coding around these parts.

I could be wrong, but this makes a lot of sense to me - given that this is what the intention is with these computations and it's not encompassing more concepts/forms of credit. Even if the displayed "upgrade credits" evolve to cover both pro-rated and some other form of credit, then I'd rather we keep the logic related to "pro-rated" separated and combined separately (if technically possible).

Reply:
Thanks for the decent discussion, @jdc91 @chriskmnds . Naming is always one of the hardest thing of our job, and this particular case of updating from "pro-rated credtis" into "upgrade credits" has taken countless attempts in the past, and has only been resolved in our hand in several iterations very lately (from pdgrnI-24d-p2, pdgrnI-2iW-p2 to pdgrnI-2nz-p2) . It looks like having positive impact on resolving customer confusion pdgrnI-2iW-p2#comment-4185.
That's why pro-rated credits can still be seen more prominently than upgrade credits internally, since the former has much longer history with us. The discussion of which term is better can easily prolong since that's why the project was so hard in the first place, but I'd prefer us to use the newly agreed name for consistency whenever possible internally or externally to avoid any discrepancy. 
Code-wise, it's common that the internal naming and the external naming differs, e.g. the Premium plan is still called `VALUE_BUNDLE` to date, we still have `olark` referred in many code even though it was the live chat service before Happychat. However, it's not common to rename it back to the old one. Thus, from my perspective I'd prefer us to stay with upgrade credits here. 

Does that make sense?

Reply:
Also, @chriskmnds could you help me point to where you found this log?

> When you initially sign up for a plan and then decide to upgrade to a higher tier within the same billing cycle, WordPress applies what's known as a "pro-rated credit" for the time unused on your current plan. This works by calculating the unused value of your current plan and then deducting that amount from the cost of the plan you're upgrading to. This is likely why you originally saw a lower price of C$276 for the Business plan.

I'm quite surprised to see that Happybot (or Odie?) sent this to our customers since that term has been fully deprecated from the support documentation. If they attempted to search for that in https://wordpress.com/support/ then they would be confused. It's something I'd like to raise to the happiness team.

Reply:
Yes I was actually involved in the change to transform this messaging from `pro-rated credits` to `upgrade credit`.

1. pdgrnI-24d-p2
2. pdgrnI-2iW-p2
3. pdgrnI-2nz-p2
4. pdgrnI-2rg-p2

We had a significant issue with this credit not being clear to users and causing considerable happiness queries. This is why after much deliberation and intense discussions (as shown in the links above) we decided to switch the term from pro-rated credits to upgrade credits. While both of these terms describe attributes of the same term (It's pro rated and it can only be used on an upgrade) the plan upgrade credit makes it much more explicit that this is not a general type of credit.

This change as well as the overall messaging resulted in a considerable decrease of support queries (pdgrnI-2iW-p2#comment-4185) 

This thread -> p1694467856357549/1694142911.437209-slack-CFFF01Q4V is exactly the example of why we need to shift from pro rated credits completely. The amount effort and confusion that causes anyone (user or coder) to explain that this credit is meant for only upgrading an account is intense. Having 2 terms after we decided to shift from a confusing term to a more aptly named term can make things even more worse.

To be blunt ... This equation here is categorically incorrect, and its exactly why we migrated from the term pro-rated credits. Because pro-rated credits incorrectly signals this to be the same as any other credit.

`(plan price + add-on price) - credits = total price`

It is to communicate aggressively and clearly that this is `A credit that can only be used for an UPGRADE` i.e. Upgrade Credit.

---


> Given this is such a peculiar specification of "upgrade credits" (which by definition is more general) 

Not necessarily, both have narrower definitions. One talks about where it came from, one talks about where it's going.

> I have seen numerous mentions

Yes of course this is the status quo, we are trying hard to shift from it, change is hard :) . And this is why its best to align with the new term even in code.

Reply:
> No, I don't agree that there is a rule that code language should be verbose and descriptive to match what language is used for users. More so if you start taking translations into account, whereby some terms don't even translate to anything less than a description or multiple English terms translating to single other-language term.

> Are we going to ban the use of "prorate" across the company because users understand a "simpler" language?

The point of an `ubiquitous language` is so that the code is inline with what the user sees. When the gap between what the user sees and what we use in our code is lesser, it reduces overall complexity and reduces time required to diagnose and fix potential bugs and ship improvement because the code is in close-sync with the problem domain. Or at least that is the idea. We write code in English and so the easiest way to tie what we ship with our code is by looking at the english version of the term in our docs and not look at translations. 

> No, I don't agree that there is a rule that code language should be verbose and descriptive to match what language is used for users

Not sure how else I can convince that we need to minimise the gap between the languages in code and what we show to users. From what I have seen, the fact that this credit can only applied to upgrade a plan has been misunderstood even among internal stakeholders. And giving weight to this terminology instead of pro-rated can reduce complexity and help developers empathise much more clearly with what the user sees which reduces the overall complexity and effort required to reason about this part of the domain.

The success metric for me personally here would be how quickly a fresh developer can understand the overall context of this area of the code. If you think having pro-rated in code and a completely different upgrade credit term in our docs and gradual changing vernacular of the support teams I guess we will have to agree to disagree.

Reply:
My concern here, and i'd rather if we dont generalize, is we are trying to ban (like literally) a valid and common technical term (from docs, discussions, and whatever else). I may be wrong, but his cannot possibly scale and cannot be the norm AFAIC.

I don't see how debugging anything around this will be a cause for issues. Maybe the opposite actually. Technically speaking, "upgrade credits" is not the same as "pro rated credits". The calculations here are about prorate -> the remaining value from the user's subscription. They are not "credits" added to an account, coming from a coupon, or due to some other discount. I am referring to this function per se - is this correct?

If we are concerned about having an equivalent term in a function name to what the user sees, then you can abstract it:

```
getPlanUpgradeCredits = () => getProRatedCredits() + otherCredits
```

But prorated should be prorated, not "planUpgradeCreditsRemaining" ( suggested above) or generalised as "planUpgradeCredits". Assuming prorate is what we are computing to begin with. This vocabulary is a standard apparently, not defacto or internal nomenclature. 

Perhaps a more geeky language and education is what's needed, not ignorance. Perhaps users themselves might appreciate this more too. "Educate, not hide"

Reply:
>  is we are trying to ban 

In that case aren't you advocating for banning the term `Upgrade Credit` that is what is already used right now in the code after all ? There is no banning here, it's just trying to decide which term to use specially since we are completely changing already checked in code. 

> The calculations here are about pro rated -> the remaining value from the user's subscription

Yes pro-rated describes what the code does, upgrade credit describes the domain driven reason for the credit. It packs more meaning and signals more knowledge instead of just explaining what the code does.

Plan Upgrade Credit means Credits that can be used ONLY for a plan upgrade. Not credits to be used for a plan upgrade.

> Perhaps a more geeky language and education is what's needed, not ignorance. Perhaps users themselves might appreciate this more too. "Educate, not hide"

This is not what our support numbers show after the change.

The facts are simple. We already had this pro-rated credit discount on our banner but we changed after extensive discussion and the numbers in support inquiries show it was a resounding success. My reason to include this term in code is so that I have been through this discussion and can understand that something that seems so simple can cause mass confusion. While the mechanics of the function pro-rated credit describe the function does the term `Upgrade Credit` more accurately aligns this term with the domain complexity involved. And it aligns with the term that is slowly adopt across support, users and stakeholders.

I disagree with this change. But I don't want to hold off this change any longer. Please go ahead with the most suitable path you see fit.

Reply:
I'm not reading the same from above, sorry.

FWIW code should be clear and discrete on what it does and returns. Generalised function names that speak a simplified language are far more problematic and the root of a lot of problems. Someone looking at this code for instance will have NO idea that it's computing prorated credits (which has a standard dictionary definition). They'd probably have a hard time figuring that out by going through the code line by line. They'd have a hard time explaining what this credit is outside of being "credit". We don't want to put developers in that position.

Not that different from renaming "Euclidean" or some other distance metric that's used to plot a straight-line distance between locations on a map as just "distance" because that's what the user sees. For all I know, an abstraction like that at the variable name would be mistake. A clear function name goes a long way to explain what a function does. Speaking context/user language in the code can bite back quickly and harshly. Like I mentioned, build an abstraction around it, but don't generalise code to avoid standard technical terms that convey precise meaning.


> The facts are simple. We already had this pro-rated credit discount on our banner but we changed after extensive discussion and the numbers in support inquiries show it was a resounding success.

I don't think the facts are simple. I barely knew what prorated meant until I bothered to look it up, because it's here in my code. Now I'm convinced it is a term that we should respect and embrace. It is 100% clear to me.

Reply:
Like I said, let's not generalize. I like and advocate for simple language. This is not simplicity though, it's obfuscating meaning and hiding implementation when it should be doing the opposite IMO.

Reply:
Hey, I would appreciate it if we did not rename variables as part of the project. Feels unrelated to the problem we are trying to solve.

Reply:
Awesome. Thanks for pitching in @dzver . Also really appreciate the insight and motivations shared by @jdc91 and @southp.

So I think it's reasonable to revert this back to "upgrade credits" for now. To my mind, "prorate" is a really nice term and part of the vocabulary used in such contexts, so something that I would expect to embrace. However, given there is a bit of history and confusion and concerns, I don't think a single simple commit should impose the direction either (like Veselin said). We've discussed this enough and I think we all learned something (like always). I'll change the naming back before merging :-)

Reply:
Thanks for following up and being open about it @chriskmnds . Since I've approved it already, I'm totally fine with you merging it direclty without another round of review after updating & fixing the merge conflicts. However, if you ever feel you'd like a confidence check, please feel free to let me know :)